src/platform/intel/cavs/pm_runtime.c

```diff
@@ -92,21 +94,19 @@ static inline void cavs_pm_runtime_en_ssp_clk_gating(uint32_t index)
 		    SHIM_CLKCTL_I2SFDCGB(index) :
 		    SHIM_CLKCTL_I2SEFDCGB(index - DAI_NUM_SSP_BASE)));
 
-	trace_event(TRACE_CLASS_POWER,
-		    "en-ssp-clk-gating index %d CLKCTL %08x",
+	trace_power("en-ssp-clk-gating index %d CLKCTL %08x",
 		    index, shim_read(SHIM_CLKCTL));
 #endif
 }
 
 #if defined(CONFIG_DMIC)
 static inline void cavs_pm_runtime_dis_dmic_clk_gating(uint32_t index)
 {
-#if defined(CONFIG_APOLLOLAKE)
+#if defined(CONFIG_APOLLOLAKE) || defined(CONFIG_CANNONLAKE)
 	(void)index;
 	shim_write(SHIM_CLKCTL, shim_read(SHIM_CLKCTL) | SHIM_CLKCTL_DMICFDCGB);
 
-	trace_event(TRACE_CLASS_POWER,
-		    "dis-dmic-clk-gating index %d CLKCTL %08x",
+	trace_power("dis-dmic-clk-gating index %d CLKCTL %08x",
 		    index, shim_read(SHIM_CLKCTL));
```

Comment:
do you need to re-read after the write? there are multiple occurrences of this code pattern, not sure if this is intended or required?

Reply:
Not required. Still it might be more reliable to trace the shim register value rather than the dsp register. Do you mean that re-reads might force and order the writes, potentially changing the behavior when traces are compiled-out?

Reply:
I can't think of a case where the reliability increases with an additional read or other code patterns that are similar in the firmware. Either we already re-read before a trace or we don't, I am not sure why we have an exception here.

Reply:
I've rewritten these parts not to re-read shim registers.